cypress/integration/Event Management/player_event_registration.js

```diff
@@ -0,0 +1,127 @@
+describe('Player Event Registration Test Suite', () => {
+    let last_url = ''
+    let event_id = 0
+    let registered_event_count = 0
+
+    before(() => {
+        cy.refreshDatabase()
+        cy.seed('FastSeeder')
+    })
+
+    beforeEach(() => {
+        cy.seederLogin()
+        cy.intercept('GET', last_url).as('last_url')
+        cy.visit(last_url)
+        cy.wait('@last_url')
+    })
+
+    it('Event Tab Button', () => {
+        cy.intercept('GET', Cypress.Laravel.route('event.index')).as('event_index')
+        cy.contains('a', 'Events').click()
+        cy.wait('@event_index')
+        last_url = Cypress.Laravel.route('event.index')
+    })
+
+    it('Event Index', () => {
+        cy.contains('div', 'Events')
+        cy.contains('Event Title')
+        cy.contains('Description')
+        cy.contains('Actions')
+        cy.contains('Rows per page:')
+    })
+
+    it('Event Filter', () => {
+        cy.intercept('GET', '**event*').as('event_filter')
+        cy.contains('Registered Events').click()
+        cy.wait('@event_filter')
+        cy.url().should('include', '?registered_only=1')
+
+        cy.contains('Registered Events').click()
+        cy.wait('@event_filter')
+        cy.url().should('include', '?registered_only=0')
+    })
+
+    it('Event Detail', () => {
+        cy.get('svg[data-testid="LogoutIcon"]')
+            .eq(0)
+            .parent()
+            .parent()
+            .invoke('attr', 'href')
+            .then((href) => {
+                event_id = parseInt(href.split('/').pop())
+            })
+            .then(() => {
+                cy.intercept(
+                    'GET',
+                    Cypress.Laravel.route('event.show').replace('{event}', event_id),
+                ).as('event_detail')
+                cy.get('svg[data-testid="LogoutIcon"]').eq(0).click()
```

Comment:
it'll honestly be better if you add `data-cy` selector to this

Reply:
I used it for the default modal button, thank you for the suggestion.